timemachine/fe/single_topology.py

```diff
@@ -1615,6 +1471,219 @@ def _setup_intermediate_chiral_bond_term(
 
         return ChiralBondRestraint(chiral_bond_idxs, np.array(chiral_bond_signs)).bind(jnp.array(chiral_bond_params))
 
+    def all_idxs_belong_to_core(self, idxs):
+        core_atoms = self.get_core_atoms()
+        return all([x in core_atoms for x in idxs])
+
+    def any_idxs_belong_to_dummy_a(self, idxs):
+        dummy_atoms = self.get_dummy_atoms_a()
+        return any([x in dummy_atoms for x in idxs])
+
+    def any_idxs_belong_to_dummy_b(self, idxs):
+        dummy_atoms = self.get_dummy_atoms_b()
+        return any([x in dummy_atoms for x in idxs])
+
+    def _chiral_volume_is_turning_on(self, idxs):
+        src_chiral_idxs = set(tuple(x) for x in self.src_system.chiral_atom.potential.idxs)
+        dst_chiral_idxs = set(tuple(x) for x in self.dst_system.chiral_atom.potential.idxs)
```

Comment:
nit: add as property since this is repeated a few times

Reply:
fixed